docs/tutorials/optimize/memory/runtime_stats.md

```diff
@@ -46,31 +74,53 @@ int main(void)
 }
 ```
 
-##### Side effects of enabling heap statistics
+#### Side effects of enabling heap statistics
 
-* An additional 8 bytes of overhead for each memory allocation.
-* The function `realloc` will never reuse the buffer it is resizing.
-* Memory allocation is slightly slower due to the added bookkeeping.
+- An additional 8 bytes of overhead for each memory allocation.
+- The function `realloc` never reuses the buffer it is resizing.
+- Memory allocation is slightly slower due to the added bookkeeping.
 
-#### Stack statistics
+### Stack statistics
 
-Stack stats provide information on the allocated stack size of a thread and the worst case stack usage. Any thread on the system can be queried for stack information.
+Stack stats provide information on the allocated stack size of a thread and the worst case stack use. You can query any thread on the system for stack information.
 
-To enable heap stats, add the command-line flag `-DMBED_STACK_STATS_ENABLED=1`.
+To enable stack stats from the command-line, add the command-line flag `-DMBED_STACK_STATS_ENABLED=1`, for example `mbed compile -DMBED_HEAP_STATS_ENABLED=1`
+
+Alternatively, to enable stack stats using `mbed_app.json`, add the following to `mbed_app.json`:
+
+```
+{
+    "macros": [
+        "MBED_STACK_STATS_ENABLED=1"
+    ],
+    ...
+}
+```
 
 There are two functions you can use to access the stack stats:
 
-* `mbed_stats_stack_get` calculates combined stack informations for all threads.
-* `mbed_stats_stack_get_each` provides stack informations for each thread separately.
+- `mbed_stats_stack_get` calculates combined stack informations for all threads.
+- `mbed_stats_stack_get_each` provides stack informations for each thread separately.
 
 <span class="notes">**Note**: These functions are available even when the stack stats are not enabled but always return zero for all fields.</span>
 
-#### Example use cases
+Both of these functions return a struct containing the following:
+
+
+|                 |                                                                           |
```

Comment:
Query: What is the empty extra row for? Can we remove it?

Reply:
Tables won't render without it. Verified with the docs engine